app/server.go

```diff
@@ -69,8 +69,9 @@ type Server struct {
 	EmailBatching    *EmailBatchingJob
 	EmailRateLimiter *throttled.GCRARateLimiter
 
-	Hubs                        []*Hub
+	hubs                        []*Hub
 	HubsStopCheckingForDeadlock chan bool
+	hubsLock                    sync.RWMutex
```

Comment:
It is idiomatic to place the mutex on top of the field protecting it. So that it becomes clear.